cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowAction.java

```diff
@@ -33,13 +33,11 @@
 public interface WorkflowAction extends Runnable {
 
   /**
-   * Provides a specification for this {@link WorkflowAction}.
+   * Configures the {@link WorkflowAction} with the given {@link WorkflowActionConfigurer}.
    *
-   * @return An instance of {@link WorkflowSpecification}.
-   * @deprecated Use {@link AbstractWorkflowAction#configure} instead.
+   * @param configurer A {@link WorkflowActionConfigurer} to configure the action
    */
-  @Deprecated
-  WorkflowActionSpecification configure();
+  void configure(WorkflowActionConfigurer configurer);
```

Comment:
It was added in the `AbstractWorkflowAction`. I am promoting it to the interface (similar thing was done in Flowlet/AbstractFlowlet).